### PR TITLE
Added 4.6 web console release notes for several items

### PR DESCRIPTION
**Pull request by ahardin-rh**
https://issues.redhat.com/browse/CONSOLE-2261
https://issues.redhat.com/browse/CONSOLE-2283
https://issues.redhat.com/browse/CONSOLE-2091
https://issues.redhat.com/browse/CONSOLE-2254

### CODE REVIEW

**Comment by ahardin-rh** [comment]
@rhamilto @spadgett @yapei PTAL Thank you!

File: release_notes/ocp-4-6-release-notes.adoc [hunk 1]
```diff
@@ -56,6 +56,44 @@ link:https://access.redhat.com/support/policy/updates/openshift[Red Hat OpenShif
...
+You can now see the schema grouping of `specDescriptor` fields and the status of
+your Operands on the Operand's details view, so that you can easily see the
+status and configure the `spec` of the Operand instance.
+
+[id="ocp-4-6-web-console-view-operators-related-objects"]
+==== View cluster Operators' related objects
+
+Previously, when viewing a cluster Operator, it was not clear what resources the
+Operator was associated with. When viewing a resource, it was not apparent that
+that resource might be related or managed by an Operator. When troubleshooting a
```
Comment (spadgett) [inline release_notes/ocp-4-6-release-notes.adoc:80]: We haven't changed the details page for resources owned by a cluster operator. I'd suggest we take the second sentence "When viewing a resource..." out.

File: release_notes/ocp-4-6-release-notes.adoc [hunk 2]
```diff
@@ -56,6 +56,44 @@ link:https://access.redhat.com/support/policy/updates/openshift[Red Hat OpenShif
...
+
+[id="ocp-4-6-web-console-improved-operand-details-view"]
+==== Improved Operand details view
+
+You can now see the schema grouping of `specDescriptor` fields and the status of
+your Operands on the Operand's details view, so that you can easily see the
+status and configure the `spec` of the Operand instance.
+
+[id="ocp-4-6-web-console-view-operators-related-objects"]
+==== View cluster Operators' related objects
```
Comment (spadgett) [inline release_notes/ocp-4-6-release-notes.adoc:76]: If we have this in the release notes, I'd also mention https://issues.redhat.com/browse/CONSOLE-2202, which adds a managed by link for operands owned by OLM operators.

**Review by spadgett — COMMENTED** [review]
Thanks @ahardin-rh. I would mention https://issues.redhat.com/browse/CONSOLE-2242 as well, which is one of the biggest console features in 4.6.

File: release_notes/ocp-4-6-release-notes.adoc [hunk 3]
```diff
@@ -56,6 +56,70 @@ link:https://access.redhat.com/support/policy/updates/openshift[Red Hat OpenShif
...
+annotation, guiding the user on what resource the user should edit.
+
+[id="ocp-4-6-web-console-specDescriptor-supports-CRD-instance"]
+==== `k8sResourcePrefix` specDescriptor supports CRD instance
+
+Operator authors, maintainers, and providers can now specify the prerequisite
+object for an Operand's instance with `Group/Version/Kind` for assigning a
+resource type other than Kubernetes core API.
+
+For more information, see
```
Comment (ahardin-rh) [inline release_notes/ocp-4-6-release-notes.adoc:120]: @sferich888 Are you okay with me including this link?

**Comment by ahardin-rh** [comment]
@spadgett Updated. Thank you!

File: release_notes/ocp-4-6-release-notes.adoc [hunk 4]
```diff
@@ -56,6 +56,70 @@ link:https://access.redhat.com/support/policy/updates/openshift[Red Hat OpenShif
...
+clickable resource link for the managing resource.
+* When the resource is modified or deleted, a message appears warning the user
+that their changes might be reverted.
+* An Operator developer can now specify a custom message to appear via an
+annotation, guiding the user on what resource the user should edit.
+
+[id="ocp-4-6-web-console-specDescriptor-supports-CRD-instance"]
+==== `k8sResourcePrefix` specDescriptor supports CRD instance
+
+Operator authors, maintainers, and providers can now specify the prerequisite
```
Comment (yapei) [inline release_notes/ocp-4-6-release-notes.adoc:116]: Before this implementation, k8sResourcePrefix specDescriptor only support k8s core API objects. So maybe we can update this sentence to 
```
Operator authors, maintainers, and providers can now specify the k8sResourcePrefix specDescriptor with `Group/Version/Kind` for assigning CRD resource type besides Kubernetes core API
```

File: release_notes/ocp-4-6-release-notes.adoc [hunk 5]
```diff
@@ -56,6 +56,70 @@ link:https://access.redhat.com/support/policy/updates/openshift[Red Hat OpenShif
...
+
+When administrators install Operators with OperatorHub, they now get immediate
+feedback to ensure that the Operator is installing properly.
+
+[id="ocp-4-6-web-console-improved-operand-details-view"]
+==== Improved Operand details view
+
+You can now see the schema grouping of `specDescriptor` fields and the status of
+your Operands on the Operand's details view, so that you can easily see the
+status and configure the `spec` of the Operand instance.
```
Comment (yapei) [inline release_notes/ocp-4-6-release-notes.adoc:84]: looks good to me

File: release_notes/ocp-4-6-release-notes.adoc [hunk 6]
```diff
@@ -56,6 +56,70 @@ link:https://access.redhat.com/support/policy/updates/openshift[Red Hat OpenShif
...
+
+[id="ocp-4-6-web-console-view-operators-related-objects"]
+==== View cluster Operators' related objects
```
Comment (bergerhoffer) [inline release_notes/ocp-4-6-release-notes.adoc:87]: Though technically fine, I'd suggest rewording to "View related objects for cluster Operators"

File: release_notes/ocp-4-6-release-notes.adoc [hunk 7]
```diff
@@ -56,6 +56,70 @@ link:https://access.redhat.com/support/policy/updates/openshift[Red Hat OpenShif
...
+
+* A *Managed by* label now appears to the right of the resource name with a
+clickable resource link for the managing resource.
+* When the resource is modified or deleted, a message appears warning the user
+that their changes might be reverted.
+* An Operator developer can now specify a custom message to appear via an
+annotation, guiding the user on what resource the user should edit.
+
+[id="ocp-4-6-web-console-specDescriptor-supports-CRD-instance"]
+==== `k8sResourcePrefix` specDescriptor supports CRD instance
```
Comment (bergerhoffer) [inline release_notes/ocp-4-6-release-notes.adoc:114]: We typically try not to start with something in backticks. Can this be reworded so that it's not first?

File: release_notes/ocp-4-6-release-notes.adoc [hunk 8]
```diff
@@ -56,6 +56,70 @@ link:https://access.redhat.com/support/policy/updates/openshift[Red Hat OpenShif
 
 This release adds improvements related to the following components and concepts.
 
+[id="ocp-4-6-web-console"]
+=== Web console
+
+[id="ocp-4-6-web-console-improved-upgrade-experience"]
+==== Improved upgrade experience in the web console
+
+* Administrators are now better informed about the differences between the upgrade
```
Comment (bergerhoffer) [inline release_notes/ocp-4-6-release-notes.adoc:65]: via -> by

File: release_notes/ocp-4-6-release-notes.adoc [hunk 9]
```diff
@@ -56,6 +56,70 @@ link:https://access.redhat.com/support/policy/updates/openshift[Red Hat OpenShif
...
+channels via helpful text and links in the web console.
+* Changes such as bug fixes and enhancements are now included in each minor and
+patch release.
+* There is now a visual representation of the different upgrade paths.
+* Special Alerts now inform administrators when new patch releases, new minor
```
Comment (bergerhoffer) [inline release_notes/ocp-4-6-release-notes.adoc:70]: alerts

File: release_notes/ocp-4-6-release-notes.adoc [hunk 10]
```diff
@@ -56,6 +56,70 @@ link:https://access.redhat.com/support/policy/updates/openshift[Red Hat OpenShif
...
+status and configure the `spec` of the Operand instance.
+
+[id="ocp-4-6-web-console-view-operators-related-objects"]
+==== View cluster Operators' related objects
+
+Previously, when viewing a cluster Operator, it was not clear what resources the
+Operator was associated with. When troubleshooting a cluster Operator, it could
+be challenging to locate the logs for all the resources that the Operator
+managed, which might be needed for troubleshooting. Now, with {product-title}
+{product-version}, you can expose a list of related objects of a Cluster
```
Comment (bergerhoffer) [inline release_notes/ocp-4-6-release-notes.adoc:93]: Cluster -> cluster

File: release_notes/ocp-4-6-release-notes.adoc [hunk 11]
```diff
@@ -56,6 +56,70 @@ link:https://access.redhat.com/support/policy/updates/openshift[Red Hat OpenShif
...
+Operator and easily review one of the related objects' details or YAML code view
```
Comment (bergerhoffer) [inline release_notes/ocp-4-6-release-notes.adoc:94]: Can "view" be removed here?

File: release_notes/ocp-4-6-release-notes.adoc [hunk 12]
```diff
@@ -56,6 +56,70 @@ link:https://access.redhat.com/support/policy/updates/openshift[Red Hat OpenShif
...
+service, or ConfigMap. Users are discouraged from editing these resources.
+Instead, users should edit the custom resources for the Operator and its
+operand, and expect the Operator to update its related resources. With this
+update:
+
+* A *Managed by* label now appears to the right of the resource name with a
+clickable resource link for the managing resource.
+* When the resource is modified or deleted, a message appears warning the user
+that their changes might be reverted.
+* An Operator developer can now specify a custom message to appear via an
```
Comment (bergerhoffer) [inline release_notes/ocp-4-6-release-notes.adoc:110]: via -> in?

File: release_notes/ocp-4-6-release-notes.adoc [hunk 13]
```diff
@@ -56,6 +56,70 @@ link:https://access.redhat.com/support/policy/updates/openshift[Red Hat OpenShif
...
+Operator and easily review one of the related objects' details or YAML code view
+for troubleshooting.
+
+[id="ocp-4-6-web-console-warning-messages-when-editing-managed-resources"]
+==== Warning messages when editing managed resources
+
+Some resources are managed, such an Operator managed by a deployment, route,
+service, or ConfigMap. Users are discouraged from editing these resources.
+Instead, users should edit the custom resources for the Operator and its
+operand, and expect the Operator to update its related resources. With this
```
Comment (bergerhoffer) [inline release_notes/ocp-4-6-release-notes.adoc:103]: Is "operand" uppercase or lower? You have sections that use it both ways

File: release_notes/ocp-4-6-release-notes.adoc [hunk 14]
```diff
@@ -56,6 +56,70 @@ link:https://access.redhat.com/support/policy/updates/openshift[Red Hat OpenShif
...
+[id="ocp-4-6-web-console-specDescriptor-supports-CRD-instance"]
+==== `k8sResourcePrefix` specDescriptor supports CRD instance
+
+Operator authors, maintainers, and providers can now specify the prerequisite
+object for an Operand's instance with `Group/Version/Kind` for assigning a
+resource type other than Kubernetes core API.
+
+For more information, see
+link:https://github.com/openshift/console/blob/master/frontend/packages/operator-lifecycle-manager/src/components/descriptors/reference/reference.md[OLM
+Descriptor Reference]
```
Comment (bergerhoffer) [inline release_notes/ocp-4-6-release-notes.adoc:122]: Needs a period

File: release_notes/ocp-4-6-release-notes.adoc (hunk 12 above)
Comment (spadgett) [reply release_notes/ocp-4-6-release-notes.adoc:110]: > via -> in?

It doesn't really appear "in" the annotation. Maybe via -> using?

File: release_notes/ocp-4-6-release-notes.adoc (hunk 13 above)
Comment (ahardin-rh) [reply release_notes/ocp-4-6-release-notes.adoc:103]: Should be lowercase. Thanks!

File: release_notes/ocp-4-6-release-notes.adoc (hunk 3 above)
Comment (ahardin-rh) [reply release_notes/ocp-4-6-release-notes.adoc:135]: @sferich888  https://issues.redhat.com/browse/OSDOCS-1475 addresses that; we can update this link once that work is complete. Please let me know if you have any concerns with that. Thanks!
